src/System.Text.Json/src/System/Text/Json/Node/JsonNumber.cs

```diff
@@ -7,89 +7,357 @@
 
 namespace System.Text.Json
 {
-    public partial class JsonNumber : JsonNode
+    public partial class JsonNumber : JsonNode, IEquatable<JsonNumber>
     {
+        private string _value;
+
         public JsonNumber() { }
-        public JsonNumber(string value) { }
-        public JsonNumber(byte value) { }
-        public JsonNumber(short value) { }
-        public JsonNumber(int value) { }
-        public JsonNumber(long value) { }
-        public JsonNumber(float value) { }
-        public JsonNumber(double value) { }
+        public JsonNumber(string value)
+        {
+            SetString(value);
+        }
+        public JsonNumber(byte value)
+        {
+            SetByte(value);
+        }
+        public JsonNumber(short value)
+        {
+            SetInt16(value);
+        }
+        public JsonNumber(int value)
+        {
+            SetInt32(value);
+        }
+        public JsonNumber(long value)
+        {
+            SetInt64(value);
+        }
+        public JsonNumber(float value)
+        {
+            SetSingle(value);
+        }
+        public JsonNumber(double value)
+        {
+            SetDouble(value);
+        }
         [CLSCompliant(false)]
-        public JsonNumber(sbyte value) { }
+        public JsonNumber(sbyte value)
+        {
+            SetSByte(value);
+        }
         [CLSCompliant(false)]
-        public JsonNumber(ushort value) { }
+        public JsonNumber(ushort value)
+        {
+            SetUInt16(value);
+        }
         [CLSCompliant(false)]
-        public JsonNumber(uint value) { }
+        public JsonNumber(uint value)
+        {
+            SetUInt32(value);
+        }
         [CLSCompliant(false)]
-        public JsonNumber(ulong value) { }
+        public JsonNumber(ulong value)
+        {
+            SetUInt64(value);
+        }
 
-        public string GetString() { throw null; }
-        public byte GetByte() { throw null; }
-        public int GetInt32() { throw null; }
-        public short GetInt16() { throw null; }
-        public long GetInt64() { throw null; }
-        public float GetSingle() { throw null; }
-        public double GetDouble() { throw null; }
+        public string GetString()
+        { 
+            return _value; 
+        }
+        public byte GetByte()
+        {
+            return byte.Parse(_value);
```

Comment:
NIT: all these methods can be written instead as oneliners which I think makes the file look cleaner:
```c#
public byte GetByte() => byte.Parse(_value);
```